x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/function/scalar/whitelist/InternalQlScriptUtils.java

```diff
@@ -51,6 +56,14 @@ public static String nullSafeSortString(Object sort) {
         return sort == null ? StringUtils.EMPTY : sort.toString();
     }
 
+    public static Number nullSafeCastNumeric(Number number, String typeName) {
+        DataType toType = fromTypeName(typeName);
+        if (toType.isNumeric() == false) {
+            throw new QlIllegalArgumentException("Casting target [" + typeName + "] is not a numerical type");
+        }
```

Comment:
Is this ever needed? Being extra careful isn't an issue however this code is called only when doing a cast so it 'seems' safe.
Furthermore, `convert` method already does safe checking so we should be fine.

Reply:
Addressed.